common/src/main/java/org/terracotta/angela/common/util/RetryUtils.java

```diff
@@ -41,20 +42,24 @@
    * @return {@code true} if the condition was evaluated to true within the given constraints, false otherwise
    */
   public static boolean waitFor(Callable<Boolean> condition, int maxWaitTimeMillis) {
-    int currRetryTime = min(maxWaitTimeMillis, 100);
-    int timeRemaining = maxWaitTimeMillis;
+    long startTime = currentTimeMillis();
+    long timeElapsed = 0;
+    long currRetryTime = min(maxWaitTimeMillis, 100);
```

Comment:
`currRetryTime` will often be 100 with a min... Are we sure this is what we want ?